### PR TITLE
KeyVerificationReady event handling

### PR DESCRIPTION
**Pull request by bitfriend**
`KeyVerificationReady` event handling was missed before.
Will implement it and close [this issue](https://github.com/acterglobal/a3/issues/789).

### CODE REVIEW

File: app/lib/features/cross_signing/cross_signing.dart [hunk 1]
```diff
@@ -735,28 +735,38 @@ class CrossSigning {
...
+    //     context: rootNavKey.currentContext!,
+    //     builder: (BuildContext context) => Container(
+    //       decoration: BoxDecoration(
+    //         borderRadius: BorderRadius.circular(15),
+    //       ),
+    //       child: _buildOnKey(context, event, flowId, emoji),
+    //     ),
+    //     isDismissible: false,
+    //   );
+    // });
```
Comment (gnunicorn) [inline app/lib/features/cross_signing/cross_signing.dart:749]: Let's just remove code we don't need anymore. We can always look it up in the git history, if needed.
```suggestion
```

File: native/acter/src/api/verification.rs [hunk 2]
```diff
@@ -960,152 +1027,362 @@ impl VerificationController {
...
+                        msg.set_content("secret".to_string(), evt.content.secret);
+                        if let Err(e) = me.event_tx.try_send(msg) {
+                            error!("Dropping transaction for {}", e);
+                        }
+                    }
+                    AnyToDeviceEvent::_Custom(evt) => {
+                        let event_type = evt.content.event_type();
+                        info!("{} got {}", device_id, event_type);
+                        warn!("{} got {}", device_id, event_type);
+                        let msg = VerificationEvent::new(
```
Comment (gnunicorn) [inline native/acter/src/api/verification.rs:1363]: more repetitions until here...

**Review by gnunicorn — APPROVED** [review]
not familiar enough to judge all that rust code, but will try it after merge.

looks good aside from two small style remarks.

File: app/lib/features/cross_signing/cross_signing.dart (hunk 1 above)
Comment (bitfriend) [reply app/lib/features/cross_signing/cross_signing.dart]: Fixed by https://github.com/acterglobal/a3/pull/883/commits/e89fe6f1714890e4363daf7b8b613663b7cb0d42

File: native/acter/src/api/verification.rs [hunk 3]
```diff
@@ -960,152 +1027,362 @@ impl VerificationController {
...
-                    .encryption()
-                    .get_device(&ev.sender, &ev.content.requesting_device_id)
-                    .await
-                {
-                    let user_id = c.user_id().expect("guest user cannot get user id");
-                    if device.user_id() == user_id && device.is_verified() {}
+                    AnyToDeviceEvent::ForwardedRoomKey(evt) => {
+                        let event_type = evt.content.event_type();
+                        info!("{} got {}", device_id, event_type);
+                        warn!("{} got {}", device_id, event_type);
```
[the remark this answers is not in the document]
Comment (bitfriend) [reply native/acter/src/api/verification.rs]: Fixed by https://github.com/acterglobal/a3/pull/883/commits/cc8896b88e05ecfbbb727433d9007af425e5ac4a

**Review by gnunicorn — APPROVED** [review]

**Comment by bitfriend** [comment]
@gnunicorn In `VerificationEvent`, `get_emojis()` replaces `async get_verification_emoji()`.
Will delete the `async get_verification_emoji()` function?

**Pull request merged**